src/Control/Monad/Catch.hs

```diff
@@ -191,13 +191,13 @@ class MonadCatch m => MonadMask m where
   generalBracket
     :: m a
     -- ^ acquire some resource
-    -> (a -> b -> m c)
+    -> (a -> m ignored1)
     -- ^ cleanup, no exception thrown
```

Comment:
Also, the Haddock field for this `(a -> m ignored1)` function is named `release` in every instance currently, although it's documented as `cleanup` (alongside another function argument which is also documented as `cleanup`). Is this intentional?